ui/src/track-layout/layout-map-api.ts

```diff
@@ -102,38 +102,73 @@ export async function getMapAlignmentsByTiles(
     changeTimes: ChangeTimes,
     mapTiles: MapTile[],
     publishType: PublishType,
-    fetchType: AlignmentFetchType,
-    selectedId?: AlignmentId,
-): Promise<AlignmentDataHolder[]> {
-    const alignmentChanged = getMaxTimestamp(
-        changeTimes.layoutLocationTrack,
-        changeTimes.layoutReferenceLine,
-    );
-    const rlChanged = getMaxTimestamp(
-        changeTimes.layoutTrackNumber,
-        changeTimes.layoutReferenceLine,
+) {
+    const polyLines = await Promise.all(
+        mapTiles.map((tile) =>
+            getPolyLines(tile, changeTimes.layoutReferenceLine, publishType, 'ALL'),
+        ),
+    ).then((p) => p.flat());
+
+    const [rlPolyLines, ltPolyLines] = partitionBy(
+        polyLines,
+        (i) => i.alignmentType === 'REFERENCE_LINE',
     );
-    const ltChanged = changeTimes.layoutLocationTrack;
-    const polyLines = (
-        await Promise.all(
-            mapTiles.map((tile) =>
-                getPolyLines(tile, alignmentChanged, publishType, fetchType, selectedId),
-            ),
-        )
-    ).flat();
-    const rlIds = getIds(polyLines, 'REFERENCE_LINE');
-    const ltIds = getIds(polyLines, 'LOCATION_TRACK');
-    const rlHeaders = await getAlignmentHeaders(publishType, rlIds, 'REFERENCE_LINE', rlChanged);
-    const ltHeaders = await getAlignmentHeaders(publishType, ltIds, 'LOCATION_TRACK', ltChanged);
-    const trackNumbers = await getTrackNumbers(publishType, changeTimes.layoutTrackNumber);
-    return combine([...rlHeaders, ...ltHeaders], polyLines, trackNumbers);
+
+    return [
+        ...(await getAlignmentDataHolder('REFERENCE_LINE', rlPolyLines, publishType, changeTimes)),
+        ...(await getAlignmentDataHolder('LOCATION_TRACK', ltPolyLines, publishType, changeTimes)),
+    ];
+}
+
+export async function getReferenceLineMapAlignmentsByTiles(
+    changeTimes: ChangeTimes,
+    mapTiles: MapTile[],
+    publishType: PublishType,
+) {
+    const polyLines = await Promise.all(
+        mapTiles.map((tile) =>
+            getPolyLines(tile, changeTimes.layoutReferenceLine, publishType, 'REFERENCE_LINES'),
+        ),
+    ).then((p) => p.flat());
+
+    return getAlignmentDataHolder('REFERENCE_LINE', polyLines, publishType, changeTimes);
+}
+
+export async function getLocationTrackMapAlignmentsByTiles(
+    changeTimes: ChangeTimes,
+    mapTiles: MapTile[],
+    publishType: PublishType,
+    locationTrackId?: LocationTrackId,
+) {
+    const polyLines = await Promise.all(
+        mapTiles.map((tile) =>
+            getPolyLines(tile, changeTimes.layoutLocationTrack, publishType, 'LOCATION_TRACKS'),
+        ),
+    )
+        .then((lines) => lines.flat())
+        .then((lines) => {
+            return lines.filter((line) => (locationTrackId ? line.id === locationTrackId : true));
```

Comment:
Tuo filter kannattanee muuten tehdä vain jos locationtrackId on olemassa, sillä se iteroi joka tapauksessa koko listan läpi ja generoi siitä duplikaatin vaikka filter ehto olisikin true. Eli ternääri mielummin jotenkin näin:
```locationTrackId ? lines.filter(line => line.id === locationTrackId) : lines```